apriltag_ros/src/apriltag_detector_node.cpp

```diff
@@ -31,14 +31,15 @@ void ApriltagDetectorNode::ImageCb(const ImageConstPtr &image_msg) {
 
   // Detect
   auto apriltags = detector_->Detect(gray);
-
+  
   // Publish apriltags
   auto apriltag_array_msg = boost::make_shared<ApriltagArrayStamped>();
   apriltag_array_msg->header = image_msg->header;
-  // this should be configurable but the idea is if they are adding more apriltags mid season we have more problems to worry about
-  // check if it is within range 
-  // @TODO make this configurable 
-  constexpr int LEGAL_TAGS[16] = {1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16};
+
+  // Load config for legal tags
+  std::vector<int> LEGAL_TAGS;
```

Comment:
I know this PR was just extending the previous code which worked with a vector.  But there's a potentially better way to do this.

C++ has a std::set data structure which is a perfect fit for this. It's a structure which simply tracks members of a set. In this case, you'd set it up with the tags IDs that are valid, and then in the detection loop just check that .count() is non-zero (it's going to be either 0 or 1, the .count() function is used by the container to be compatible with other data structures where count can be any value).

This set will take a slight bit of extra work to set up (read the param into a vector, then loop to .emplace() each entry from that vector into the set) but the lookup will be quicker and easier to read.

Speaking of which, it looks like the code currently re-reads the param each time an image is processed. Instead, this param read should be made part of the ApriltagDetectorNode constructor and the set data structure stored as a class member.   No need to re-read the config value every image when it won't change.

Reply:
Added the suggested changes to apriltag_detector_node -- it seems to be working based on my testing -- and currently working on the GPU port. Thanks so much for your help!